### ISSUE TITLE
Notes with brackets in their names trigger parse errors

### ISSUE THREAD

**Issue by dmn-bernard** [javalent/dice-roller#131]
Hi and first of all thanks for this awesome plugin! ❤️ 

I've been struggling for a moment as to why I couldn't make my look up table work properly. Then I realised after many tests that it was due to the brackets in the name of my files. Let me give you a few screenshots as explanations:

This is an attempt without bracket (edition mode)
![Capture d’écran 2022-07-01 à 19 26 26](https://user-images.githubusercontent.com/12032776/176942547-1e652b78-5b21-4b78-9e57-80e95ac71cf9.png)

It then works correctly when rolling 👍 
![Capture d’écran 2022-07-01 à 19 26 21](https://user-images.githubusercontent.com/12032776/176942588-3995b8dd-1cc5-485d-8538-d28b8982c771.png)

However if I add parenthesis in my file name, I will get a parsing error
![Capture d’écran 2022-07-01 à 19 27 17](https://user-images.githubusercontent.com/12032776/176942647-60f42b11-975c-4bf1-9492-5b7d82efd2ba.png)

This is not a big problem for me, I can just change the names of my files. In case this is not a solvable bug, I think it should still be specified somewhere in the documentation, or maybe inside the error, so we don't waste too much time looking for what's wrong with our implementation of our dice rollers 😅

**Comment by sigrunixia** [comment]
As a rule of thumb, you do not want special characters besides dash and underscore within your file names for this reason. The fact that Obsidian allows it, has long been the bane of plugin developers. 

Closing this thread.

**Comment by dmn-bernard** [comment]
Yeah I understand, a note in the documentation still couldn't hurt the plugin users. Maybe it's a well known fact among plugin developers, but probably less among their users.

**Comment by valentine195** [comment]
I think this is just a regex issue, I will take a look c

**Comment by zschoenstadt** [comment]
Parens are very much legal in most file name standards. This is due to an uneccessary stripping of Parens in two of the rollers, after they had already been regex parsed with proper paren and bracket handling.
